modules/serverless-kafka-sasl.adoc

```diff
@@ -0,0 +1,83 @@
+// Module is included in the following assemblies:
+//
+// serverless/serverless-kafka.adoc
+
+[id="serverless-configuring-sasl-authentication-against-an-apache-kafka_{context}"]
+== Configuring SASL authentication
+
+.Prerequisites
+
+* A username and password for the Kafka cluster.
+* Choose the SASL mechanism to use, for example `PLAIN`, `SCRAM-SHA-256` or `SCRAM-SHA-512`.
+* If TLS is enabled, you also need the `ca.crt` certificate file for the Kafka cluster.
+
+Red Hat recommends to enable TLS in addition to SASL.
+
+.Procedure
+
+. Create the certificate files as secrets in your chosen namespace:
++
+----
+$ kubectl create secret --namespace <namespace> generic <kafka-auth-secret> \
+  --from-file=ca.crt=caroot.pem \
+  --from-literal=password="SecretPassword" \
+  --from-literal=saslType="SCRAM-SHA-512" \
+  --from-literal=user="my-sasl-user"
+----
++
+[IMPORTANT]
+====
+Use the key names `ca.crt`, `password`, and `saslType`. Do not change them.
+====
+
+. Start editing the `KnativeKafka` CR:
++
+----
+$ oc edit knativekafka
+----
+
+. Reference your secret and the namespace of the secret:
++
+----
+apiVersion: operator.serverless.openshift.io/v1alpha1
+kind: KnativeKafka
+metadata:
+  namespace: knative-eventing
+  name: knative-kafka
+spec:
+  channel:
+    authSecretName: <kafka-auth-secret>
+    authSecretNamespace: <kafka-auth-secret-namespace>
+    bootstrapServers: <bootstrap-server>
+    enabled: true
+  source:
+    enabled: true
+----
++
+[NOTE]
+====
+Make sure to specify the matching port in the bootstrap server.
+====
++
+For example:
++
+----
+apiVersion: operator.serverless.openshift.io/v1alpha1
+kind: KnativeKafka
+metadata:
+  namespace: knative-eventing
+  name: knative-kafka
+spec:
+  channel:
+    authSecretName: scram-user
+    authSecretNamespace: kafka
+    bootstrapServers: eventing-kafka-bootstrap.kafka.svc:9092
```

Comment:
```suggestion
    bootstrapServers: eventing-kafka-bootstrap.kafka.svc:9093
```

Let's use 9093 here since that's the traditional SASL port for Kafka.